src/compiler/transformers/es2015.ts

```diff
@@ -1089,19 +1099,77 @@ namespace ts {
                 insertCaptureThisForNodeIfNeeded(prologue, constructor);
             }
 
-            const block = factory.createBlock(
+            const bodyBlock = factory.createBlock(
                 setTextRange(
                     factory.createNodeArray(
-                        concatenate(prologue, statements)
+                        [
+                            ...existingPrologue,
+                            ...prologue,
+                            ...visitNodes(factory.createNodeArray(preSuperStatements), visitor, isStatement),
+                            ...statements
+                        ]
                     ),
                     /*location*/ constructor.body.statements
                 ),
                 /*multiLine*/ true
             );
 
-            setTextRange(block, constructor.body);
+            setTextRange(bodyBlock, constructor.body);
+            return bodyBlock;
+        }
 
-            return block;
+        function splitConstructorBodyStatementTypes(originalBodyStatements: NodeArray<Statement>) {

```

Comment:
I think this is introducing too much complexity. We probably need to inline this into `splitConstructorBodyStatementTypes` and find ways of simplifying the behavior. I've added a similar comment in _ts.ts_.